engine/src/components/component.h

```diff
@@ -28,34 +28,64 @@
 #ifndef COMPONENT_H
 #define COMPONENT_H
 
+#include <string>
+#include <map>
+
+/**
+ * LibComponent is currently tightly coupled to LibDamage and
+ * other various libraries in VegaStrike engine.
+ * Consider decoupling and subclassing every component in it.
+ */
+
+class Unit;
+
+// TODO: add complete list
+enum class ComponentType {
+    Hull, 
+    Armor, 
+    Shield,
+    Drive
+};
 
 class Component
 {
 protected:
-    bool installed = true;
-    bool enabled = true;
-    bool damaged = false;
+    std::string upgrade_name;   // Isometal Armor
+    std::string upgrade_key;    // armor03__upgrades
+    
+    double mass = 0;
+    double volume = 0;
+
+    bool integral = false; // Part of the ship. Can't be upgraded/downgraded
 public:
-    Component();
-    virtual ~Component() {}
+    Component(std::string upgrade_name, double mass, 
+              double volume, bool integral);
 
-    void Install() { installed = true; }
-    void Uninstall() { installed = false; }
-    bool Installed() { return installed; }
+    // Load from units dictionary
+    virtual void Load(std::string upgrade_key, std::string unit_key);      
+    
+    virtual void SaveToCSV(std::map<std::string, std::string>& unit) const = 0;
```

Comment:
is this doing something specific to CSV files? Or does it work for JSON and other formats too?
E.g could we just call it `Save`? Or does it need to only apply to CSV files?

Reply:
All our saves are CSV only.

Reply:
It'd be nice to change that up. I'd like to change the saved game format to use JSON too, perhaps also use a zip like format similar to ODF. But that's another refactor

Reply:
Since CSV is an implementation detail that could change, it's probably better to describe it more generically. I think `Save` or `Saves` is great, personally.